beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -115,8 +115,27 @@ func (f *ForkChoice) InsertNode(ctx context.Context, state state.BeaconState, ro
 		return errNilBlockHeader
 	}
 	parentRoot := bytesutil.ToBytes32(bh.ParentRoot)
-	var payloadHash [32]byte
-	if state.Version() >= version.Bellatrix {
+	var payloadHash, parentHash [32]byte
+	if state.Version() >= version.EPBS {
+		slot, err := state.LatestFullSlot()
+		if err != nil {
+			return err
+		}
+		latestHash, err := state.LatestBlockHash()
+		if err != nil {
+			return err
+		}
+		if slot == state.Slot() {
+			latestHeader, err := state.LatestExecutionPayloadHeaderEPBS()
+			if err != nil {
+				return err
+			}
+			copy(payloadHash[:], latestHash)
+			copy(parentHash[:], latestHeader.ParentBlockHash)
+		} else {
+			copy(parentHash[:], latestHash)
```

Comment:
If the incoming state is full (as will be when processing a `SignedExecutionPayloadEnvelope`) we can pass a non-trivial `payloadHash` and a nontrivial `parentHash`. This is the first branch of the if-else highlighted in this comment. If the incoming state is not full (that is when we process a `BeaconBlock`) then we can only pass the `parentHash` and pass a zero payloadHash to the store.